hibernate-core/src/test/java/org/hibernate/test/hql/NullSelectExpressionTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.hql;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.hql.internal.ast.QuerySyntaxException;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * @author Gail Badner
+ */
+@TestForIssue( jiraKey = "HHH-10757")
+public class NullSelectExpressionTest extends BaseCoreFunctionalTestCase {
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		Object result = s.createQuery(
+				"select firstName, null, lastName from NullSelectExpressionTest$Person where lastName='Munster'"
+		).uniqueResult();
```

Comment:
The following fails because Hibernate assumes there was only 1 select expression returned by the query. The null select expression seems to be interpreted as "no more select expressions". The query returns a String object instead Object[].